.github/workflows/publish.yaml

```diff
@@ -0,0 +1,61 @@
+name: Publisher
+on: 
+  push:
+    branches:
+      - main
+
+jobs:
+  semver:
+    name: Generate semantic version number
+    outputs:
+      semantic_version: ${{ steps.semver.outputs.semantic_version }}
+    runs-on: ubuntu-latest
+    steps:  
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+      - id: semver
+        uses: lukaszraczylo/semver-generator@1.4.18
+        with:
+          config_file: .github/semver.yaml
+          repository_local: true
+      - uses: marvinpinto/action-automatic-releases@v1.2.1
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          automatic_release_tag: ${{ steps.semver.outputs.semantic_version}}
+          prerelease: false
+          title: ${{ steps.semver.outputs.semantic_version}}
+          files: "*"
+
+  arduino:
```

Comment:
I recommend adding an `environment:` and keeping secrets under that rather than repo-global.

Reply:
updated to use platformio_publish environment